File: SimpleRT/cpp/Shape3D.cpp
// ====================================================================================================
// Produced by the Free Edition of Java to C++ Converter.
// To produce customized conversions, purchase a Premium Edition license:
// https://www.tangiblesoftwaresolutions.com/product-details/java-to-cplus-converter.html
// ====================================================================================================

#include "Shape3D.h"

void Shape3D::UpdateAttenuation(std::shared_ptr<RGBColorFloat> attenuation)
{
   attenuation->r *= material->ktCr;
   attenuation->g *= material->ktCg;
   attenuation->b *= material->ktCb;
}

double Shape3D::GetCenterCoord(int axis)
{
   switch (axis)
   {
	  case 0 :
		  return center->x;
	  case 1 :
		  return center->y;
	  case 2 :
		  return center->z;
   }
   return 0.0;
}

File: SimpleRT/src/Shape3D.java

public abstract class Shape3D {
   public int mat_index;    
   public int part_index;
   public MatAttr  material;
   public RGBColor native_color;   
   public BoundingVolume  bv;
   public Point3D  center;
   public abstract boolean IsIntersected( RTRay ray );    
   public abstract void GetNormal( Point3D p, Vector3D N );  
   
   public void UpdateAttenuation( RGBColorFloat attenuation )
   {
      attenuation.r *= material.ktCr;
      attenuation.g *= material.ktCg;
      attenuation.b *= material.ktCb;      
   }
   
   public double GetCenterCoord( int axis )
   {
      switch (axis )  {
         case 0 : return center.x;
         case 1 : return center.y;
         case 2 : return center.z;
      }      
      return 0.0;
   }
}
